snippets/product-thumbnail.liquid

```diff
@@ -51,8 +50,8 @@
 {%- endcapture -%}
 
 <div
-  class="product-media-container gradient global-media-settings"
-  style="--ratio-percent: {{ preview_media_ratio }}%;"
+  class="product-media-container media-type-{{ media.media_type }} media-fit-{{ media_fit }}{% if constrain_to_viewport %} constrain-height{% endif %} gradient global-media-settings"
```

Comment:
+1 :D

Reply:
We never aligned on this formally as a team. I don't feel too strongly about pushing back in this particular case, so I can adjust, but I'm not always sold on basing the order of classes because of the arbitrary decision to include an inline conditional or not. `gradient` for a example is a much more generic global class and we usually like to order those after more specific classes. Again it's a bit of a toss up here if gradient and media-settings are more generic than constrain-height, but that's how I think about it.

```class=”grid grid--2-col has-border show-heading-or-something slider {% if show_slider_desktop %} slider–desktop{% endif %}{% if some_setting %} grid--1-col{% endif %}”```

Also consider this ^ example, `grid--1-col` gets moved to the end and is now harder to associate with other relevant related classes. I don't like that and that doesn't make this code any easier to understand at a glance.